parquet-column/src/main/java/org/apache/parquet/schema/PrimitiveType.java

```diff
@@ -222,6 +266,11 @@ public void addValueToPrimitiveConverter(
       public <T, E extends Exception> T convert(PrimitiveTypeNameConverter<T, E> converter) throws E {
         return converter.convertINT96(this);
       }
+
+      @Override
+      PrimitiveComparator<?> comparator(OriginalType logicalType) {
+        return PrimitiveComparator.SIGNED_BINARY_COMPARATOR;
```

Comment:
This should be undefined according to [PARQUET-1065](https://issues.apache.org/jira/browse/PARQUET-1065).

Reply:
Implementing undefined comparator logic is not a simple effort and a bit error prone. I can see two options:
* Returning `null` wherever we provide comparators and handle `null` comparator in statistics logic to not to update min-max neither load or save it. In this case `NullPointerException` might occur in client code which is misleading.
* Returning a special implementation of the `PrimitiveComparator` which throws proper exceptions in case of `compare` is invoked. (Also update the statistics logic.) In this case it might not be obvious for the client code to determine whether the comparator is undefined or not.
I would prefer the latter one. What do you think?

Reply:
In my opinion the best approach would be to throw a checked exception. This way API consumers will be forced to handle the lack of comparators. This is important, because it can happen for unsupported logical types as detailed in my previous comment.

Reply:
I think this should return the comparator that is documented in logical types for consistency. We should just discard statistics for these types when reading and should not produce statistics when writing.